### PR TITLE
Fix #932 NullPointerException when parsing 3FDJ COMPND records

### LINKED ISSUE #932: Empty entity list when parsing PDB ID 3FDJ source record
When I tried to parse 3FDJ, an IndexOutOfBoundsException is thrown in the pdb_SOURCE_Handler() function.

> Exception in thread "main" java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
	at java.base/java.util.ArrayList.get(ArrayList.java:427)
	at org.biojava.nbio.structure.io.PDBFileParser.sourceValueSetter(PDBFileParser.java:1255)
	at org.biojava.nbio.structure.io.PDBFileParser.pdb_SOURCE_Handler(PDBFileParser.java:1221)
	at org.biojava.nbio.structure.io.PDBFileParser.makeCompounds(PDBFileParser.java:2755)
	at org.biojava.nbio.structure.io.PDBFileParser.parsePDBFile(PDBFileParser.java:2695)
	at org.biojava.nbio.structure.io.PDBFileParser.parsePDBFile(PDBFileParser.java:2557)
	at org.biojava.nbio.structure.io.PDBFileReader.getStructure(PDBFileReader.java:140)

It seems to be more related to the entities rather than the SOURCE records.
Below is a bug generating code snippet.
```
	private static void demoAtomCache() {
		UserConfiguration userConfiguration = new UserConfiguration();
		userConfiguration.setFileFormat(UserConfiguration.PDB_FORMAT);
		AtomCache cache = new AtomCache(userConfiguration);

		try {
			Structure s = cache.getStructure("3fdj");
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
```

### PR DESCRIPTION
**Pull request by aalhossary**
The problem is because a colon (:) is present in the first token and it
was not properly handled.
Fixed #932

### CODE REVIEW

**Comment by josemduarte** [comment]
@aalhossary , we are having some trouble with the CI tests. Could you try pushing an empty commit (`git commit --allow-empty -m "Trigger Build"
`) to see if that triggers the tests?

File: biojava-structure/src/test/resources/1hhbCMPND+SRC.ent [hunk 1]
```diff
@@ -0,0 +1,18 @@
+HEADER    OXYGEN TRANSPORT                        01-APR-75   1HHB
```
Comment (josemduarte) [inline biojava-structure/src/test/resources/1hhbCMPND+SRC.ent:1]: Could you place these new resources files in dir `src/test/resources/org/biojava/nbio/structure/io` ?

**Review by josemduarte — CHANGES_REQUESTED** [review]
Thank you! One comment about where to place resource files

File: biojava-structure/src/test/resources/1hhbCMPND+SRC.ent (hunk 1 above)
Comment (aalhossary) [reply biojava-structure/src/test/resources/1hhbCMPND+SRC.ent]: Done

**Review by josemduarte — APPROVED** [review]
Thanks! LGTM

Regarding CI: travis seems to be working fine now. But the github workflow somehow seems to stall. Since you added the configuration for that @HanSolo, would you know what might be happening?

**Pull request merged**